docs/dev/tombstone.md

```diff
@@ -0,0 +1,572 @@
+-------------------+-----------------+------------------+-----+-----+-----------------+-------------------------------------------------------------------------------------+------------------------+-------
+ range tombstone 1 |      memtable:0 |                0 |     |     |                 |                                                                    {"tombstone":{}} |        partition start |  null
+ range tombstone 1 |      memtable:0 |                2 |   0 | 100 |               1 | {"tombstone":{"timestamp":1743055013006807,"deletion_time":"2025-03-27 05:56:53z"}} | range tombstone change |  null
+ range tombstone 1 |      memtable:0 |                2 |   0 | 200 |              -1 |                                                                    {"tombstone":{}} | range tombstone change |  null
+ range tombstone 1 |      memtable:0 |                3 |     |     |                 |                                                                                null |          partition end |  null
+
+(4 rows)
+```
+
+Internally, range tombstones are represented by so-called "range tombstone change" objects.
+There are two such objects: one at the start of the range tombstone (the lower bound key) and one at the end of the range tombstone (the higher bound key).
+The range tombstone end object resets the tombstone to the null-tombstone (no deletion).
+
+Delete a prefix -- only a prefix of all the clustering columns is restricted:
+```CQL
+DELETE FROM ks.tbl WHERE pk = 'range tombstone 2' AND ck1 = 1;
+
+SELECT * FROM MUTATION_FRAGMENTS(ks.tbl) WHERE pk = 'range tombstone 2';
+
+ pk                | mutation_source | partition_region | ck1 | ck2 | position_weight | metadata                                                                            | mutation_fragment_kind | value
+-------------------+-----------------+------------------+-----+-----+-----------------+-------------------------------------------------------------------------------------+------------------------+-------
+ range tombstone 2 |      memtable:0 |                0 |     |     |                 |                                                                    {"tombstone":{}} |        partition start |  null
+ range tombstone 2 |      memtable:0 |                2 |   1 |     |              -1 | {"tombstone":{"timestamp":1743055505954714,"deletion_time":"2025-03-27 06:05:05z"}} | range tombstone change |  null
+ range tombstone 2 |      memtable:0 |                2 |   1 |     |               1 |                                                                    {"tombstone":{}} | range tombstone change |  null
+ range tombstone 2 |      memtable:0 |                3 |     |     |                 |                                                                                null |          partition end |  null
+
```

Comment:
I think discussion of position_weight would be useful here.

Reply:
Done in v2.